packages/vit-s/src/hooks.js

```diff
@@ -322,3 +322,58 @@ export function useGetObsInfo(obsType, obsLabelsTypes, obsLabelsData, obsSetsMem
     return null;
   }, [obsType, obsLabelsTypes, obsLabelsData, obsSetsMembership]);
 }
+
+
+export const useMappedGeneList = (geneObject, featureMap) => {
+  const [mappedGeneList, setMappedGeneList] = useState(geneObject);
+  useEffect(() => {
+    const processFeatureLabelMap = async () => {
+      // Assuming featureMap is set for symbolic genes
+      if (featureMap !== undefined) {
+        setMappedGeneList(geneObject);
+        return;
+      }
+      try {
+        let ensbGenes = false;
+        if (Array.isArray(geneObject)) ensbGenes = geneObject?.some(value => value?.toUpperCase().startsWith('ENSG'));
+        else {
+          ensbGenes = Object.entries(geneObject)?.some(([key, value]) => key.toLowerCase().includes('gene') && value?.toUpperCase().startsWith('ENSG'));
+        }
+        // console.log(ensbGenes)
+        if (!ensbGenes) return;
+
+        let updatedGeneObject = { ...geneObject };
+        const response = await fetch('https://vitessce-resources.s3.us-east-2.amazonaws.com/genes_filtered.json');
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        const jsonData = await response.json();
+
+        if (Array.isArray(geneObject)) {
+          updatedGeneObject = geneObject.map((gene) => {
+            if (gene.toUpperCase().startsWith('ENSG')) {
+              const trimmedValue = gene.split('.')[0]; // Trim after the dot
+              return jsonData[trimmedValue];
+            }
+            return null;
+          });
+          updatedGeneObject = updatedGeneObject.filter(Boolean);
+        } else {
+          Object.entries(geneObject).forEach(([key, value]) => {
+            if (key.toLowerCase().includes('gene') && value?.toUpperCase().startsWith('ENSG')) {
+              const trimmedValue = value.split('.')[0];
+              updatedGeneObject[key] = jsonData[trimmedValue];
+            }
+          });
+        }
+        setMappedGeneList(updatedGeneObject);
+      } catch (err) {
+        /* eslint-disable-next-line no-console */
+        console.log(err.message);
```

Comment:
Would this error be worth surfacing as a warning? https://github.com/vitessce/vitessce/blob/3287daea1dfb7443430f144f7a7aa8e61180d7fb/packages/vit-s/src/state/hooks.js#L1153